datafusion/core/src/physical_plan/filter.rs

```diff
@@ -231,6 +246,38 @@ impl RecordBatchStream for FilterExecStream {
     }
 }
 
+/// Return the equals Column-Pairs and Non-equals Column-Pairs
+fn collect_columns_from_predicate(predicate: &Arc<dyn PhysicalExpr>) -> EqualAndNonEqual {
```

Comment:
Perhaps this would be better in utils.rs

Reply:
Since this is only used by FilterExec, I would prefer to keep this as a private func in filter.rs